File: app/src/main/java/com/supagorn/devpractice/ui/category/viewpager/SampleFragment.java
package com.supagorn.devpractice.ui.category.viewpager;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.supagorn.devpractice.R;

public class SampleFragment extends Fragment {
	// Store instance variables
	private String title;
	private int page;

	// newInstance constructor for creating fragment with arguments
	public static SampleFragment newInstance(int page, String title) {
		SampleFragment fragmentFirst = new SampleFragment();
		Bundle args = new Bundle();
		args.putInt("someInt", page);
		args.putString("someTitle", title);
		fragmentFirst.setArguments(args);
		return fragmentFirst;
	}

	// Store instance variables based on arguments passed
	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		page = getArguments().getInt("someInt", 0);
		title = getArguments().getString("someTitle");
	}

	// Inflate the view for the fragment based on layout XML
	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container,
							 Bundle savedInstanceState) {
		View view = inflater.inflate(R.layout.fragment_viewpager, container, false);
		TextView tvLabel = (TextView) view.findViewById(R.id.tvLabel);
		tvLabel.setText(page + " -- " + title);
		return view;
	}
}

File: app/src/main/java/com/supagorn/devpractice/ui/SampleFragment.kt
package com.supagorn.devpractice.ui

import android.os.Bundle
import com.supagorn.devpractice.R
import com.supagorn.devpractice.customs.AbstractFragment

/**
 * Created by apple on 2/17/2018 AD.
 */

class SampleFragment : AbstractFragment() {

    private var mTitle: String? = null

    companion object {
        private const val ARG_TITLE = "ARG_TITLE"
        fun newInstance(title: String): SampleFragment {
            val fragment = SampleFragment()
            val bundle = Bundle()
            bundle.putString(ARG_TITLE, title)
            fragment.arguments = bundle
            return fragment
        }
    }

    override fun setLayoutView(): Int {
        return R.layout.fragment_sample
    }

    override fun setupView() {
        //set title toolbar
        mTitle = arguments?.getString(ARG_TITLE)
        if(mTitle != null) {
            setTitle(mTitle!!)
        }
    }
}
